### PR TITLE
ReadyCallback should be a repeating callback

### PR DESCRIPTION
**Pull request by bridiver**
fix https://github.com/brave/brave-browser/issues/10271

Crash introduced when repeating callback was incorrectly changed to OnceCallback https://github.com/brave/brave-core/commit/1f1468fa32469ba47bffe321c227d93345eb0321#diff-2330ee7fe08aa6ad90e445617c0f3dafR21
and then https://github.com/brave/brave-core/pull/2526/files fixed the crash, but the callback should have been a RepeatingCallback

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/component_updater/brave_component_installer.cc [hunk 1]
```diff
@@ -124,17 +124,9 @@ void BraveComponentInstallerPolicy::ComponentReady(
...
-  // It appears to be possible for the ComponentInstaller to call
-  // `ComponentReady` more than once. There is a call in
-  // ComponentInstaller::FinishRegistration and another one in
-  // ComponentInstaller::Install. So a call to Register followed by a call
-  // to Install could result in a crash here. See
-  // https://github.com/brave/brave-browser/issues/4624
-  if (!ready_callback_.is_null()) {
-    std::move(ready_callback_).Run(
+  ready_callback_.Run(
           install_dir,
```
Comment (iefremov) [inline browser/component_updater/brave_component_installer.cc:128]: bad spacing

File: components/brave_component_updater/browser/brave_component.cc [hunk 2]
```diff
@@ -31,24 +33,33 @@ void BraveComponent::Register(const std::string& component_name,
...
+  VLOG(2) << "unregister component: " << component_id_;
   return delegate_->Unregister(component_id_);
 }
 
 scoped_refptr<base::SequencedTaskRunner> BraveComponent::GetTaskRunner() {
   return delegate_->GetTaskRunner();
 }
 
+void BraveComponent::OnComponentReadyInternal(
+  const std::string& component_id,
```
Comment (iefremov) [inline components/brave_component_updater/browser/brave_component.cc:56]: bad spacing

File: components/brave_component_updater/browser/brave_component.cc [hunk 3]
```diff
@@ -31,24 +33,33 @@ void BraveComponent::Register(const std::string& component_name,
...
 bool BraveComponent::Unregister() {
+  VLOG(2) << "unregister component: " << component_id_;
   return delegate_->Unregister(component_id_);
 }
 
 scoped_refptr<base::SequencedTaskRunner> BraveComponent::GetTaskRunner() {
   return delegate_->GetTaskRunner();
 }
 
+void BraveComponent::OnComponentReadyInternal(
```
Comment (iefremov) [inline components/brave_component_updater/browser/brave_component.cc:55]: why do we need this function?

Comment (bridiver) [reply components/brave_component_updater/browser/brave_component.cc:55]: for consistent debug logging

**Comment by bridiver** [comment]
unrelated test failure in CI

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 10.15.5 x64` using the following build:

```
Brave | 1.12.47 Chromium: 83.0.4103.116 (Official Build) nightly (64-bit)
--- | ---
Revision | 8f0c18b4dca9b6699eb629be0f51810c24fb6428-refs/branch-heads/4103@{#716}
OS | macOS Version 10.15.5 (Build 19F101)
```

* launched using `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --enable-logging=stderr --v=0 --vmodule=brave_component=2 --use-dev-goupdater-url`

```
[51766:775:0624/001524.487341:VERBOSE2:brave_component.cc(59)] component ready: {
   "data_file_version": "1",
   "description": "Brave SpeedReader Updater extension",
   "key": "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA3j/+grwCsrYVA99oDHa+E9z5edPIV3J+lzld3X7K8wfJXbSauGf2DSxW0UEh+MqkkcIK/66Kkd4veuWqnUCAGXUzrHVy/N6kksDkrScOlpKT9zfyIvLc/4nmiyPCSc5c7UrDVUwZnIUBBpEHiwkpiM4pujeJkZSl5783RWIDRN92GDBdHMdD97JH3bPp3SCTmfAAHzzYUAHUSrOAfodD8qWkfWT19VigseIqwK6dH30uFgaZIOwU9uJV2Ts/TDEddNv8eV7XbwQdL1HUEoFj+RXDq1CuQJjvQdc7YRmy0WGV0GIXu0lAFOQ6D/Z/rjtOe//2uc4zIkviMcUlrvHaJwIDAQAB",
   "manifest_version": 2,
   "name": "Brave SpeedReader Updater",
   "version": "1.0.7"
}
```

* published new component via https://ci.brave.com/view/all/job/brave-core-ext-speedreader-data-files-update-publish-dev/7/
* waited ~20min and pulled new component via `brave://components`

```
[51766:775:0624/003721.945069:VERBOSE2:brave_component.cc(59)] component ready: {
   "data_file_version": "1",
   "description": "Brave SpeedReader Updater extension",
   "key": "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA3j/+grwCsrYVA99oDHa+E9z5edPIV3J+lzld3X7K8wfJXbSauGf2DSxW0UEh+MqkkcIK/66Kkd4veuWqnUCAGXUzrHVy/N6kksDkrScOlpKT9zfyIvLc/4nmiyPCSc5c7UrDVUwZnIUBBpEHiwkpiM4pujeJkZSl5783RWIDRN92GDBdHMdD97JH3bPp3SCTmfAAHzzYUAHUSrOAfodD8qWkfWT19VigseIqwK6dH30uFgaZIOwU9uJV2Ts/TDEddNv8eV7XbwQdL1HUEoFj+RXDq1CuQJjvQdc7YRmy0WGV0GIXu0lAFOQ6D/Z/rjtOe//2uc4zIkviMcUlrvHaJwIDAQAB",
   "manifest_version": 2,
   "name": "Brave SpeedReader Updater",
   "version": "1.0.8"
}
```

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 10.15.5 x64` using the following build:

```
Brave | 1.12.47 Chromium: 83.0.4103.116 (Official Build) nightly (64-bit)
--- | ---
Revision | 8f0c18b4dca9b6699eb629be0f51810c24fb6428-refs/branch-heads/4103@{#716}
OS | macOS Version 10.15.5 (Build 19F101)
```

* launched using `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --enable-logging=stderr --v=0 --vmodule=brave_component=2 --use-dev-goupdater-url`

```
[51766:775:0624/001524.487341:VERBOSE2:brave_component.cc(59)] component ready: {
   "data_file_version": "1",
   "description": "Brave SpeedReader Updater extension",
   "key": "...",
   "manifest_version": 2,
   "name": "Brave SpeedReader Updater",
   "version": "1.0.7"
}
```

* published new component via https://ci.brave.com/view/all/job/brave-core-ext-speedreader-data-files-update-publish-dev/7/
* waited ~20min and pulled new component via `brave://components`

```
[51766:775:0624/003721.945069:VERBOSE2:brave_component.cc(59)] component ready: {
   "data_file_version": "1",
   "description": "Brave SpeedReader Updater extension",
   "key": "...",
   "manifest_version": 2,
   "name": "Brave SpeedReader Updater",
   "version": "1.0.8"
}
```

**Pull request merged**